movici_simulation_core/storage/sqlite_schema.py

```diff
@@ -142,6 +143,25 @@ class UpdateAttribute(Base):
         Integer, ForeignKey("attribute_data.id", ondelete="CASCADE"), primary_key=True
     )
 
+    # Relationships
+    update = relationship("Update", overlaps="attributes,updates")
+    attribute_data = relationship("AttributeData", overlaps="attributes,updates")
+
+
+class InitialDataset(Base):
+    """Stores initial dataset snapshots for self-contained database archives.
+
+    This allows the database to be a complete simulation record without
+    requiring separate init_data directory.
+    """
+
+    __tablename__ = "initial_dataset"
+    __table_args__ = (Index("idx_initial_dataset_name", "dataset_name"),)
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    dataset_name = Column(String, unique=True, nullable=False)
+    data = Column(LargeBinary, nullable=False)  # Serialized JSON data
```

Comment:
I think we should differentiate three different types of dataset data

1. entity oriented json data: we should destructure them into numpy arrays and store them similar to how we store updates. This way we can use it movici viewer to easily query specific attributes
2. unstructured json data: we can store them as either jsonb or a blob but it is clear that they are json loadable
3. binary data: we store them as blob and pass them transparently to consumers. 

In the platform we use the key `"format"` to describe whether a dataset contains `entity_based`, `unstructured` or `binary` data. I think would be good to adopt those terms (perhaps create an enum for it)


We may need to think about not storing a blob but a reference to a file, but I think that's something for the future